core/src/main/java/org/apache/accumulo/core/fate/accumulo/AccumuloStore.java

```diff
@@ -0,0 +1,297 @@
+          return repo;
+        }).findFirst().orElse(null);
+      });
+    }
+
+    @Override
+    public List<ReadOnlyRepo<T>> getStack() {
+      verifyReserved(false);
+
+      return scanTx(scanner -> {
+        scanner.setRange(getRow(tid));
+        scanner.fetchColumnFamily(RepoColumnFamily.NAME);
+        return StreamSupport.stream(scanner.spliterator(), false).map(e -> {
+          @SuppressWarnings("unchecked")
+          var repo = (ReadOnlyRepo<T>) deserialize(e.getValue().get());
+          return repo;
+        }).collect(Collectors.toList());
+      });
+    }
+
+    @Override
+    public Serializable getTransactionInfo(TxInfo txInfo) {
+      verifyReserved(false);
+
+      try (Scanner scanner = context.createScanner(tableName, Authorizations.EMPTY)) {
+        scanner.setRange(getRow(tid));
+
+        final ColumnFQ cq;
+        switch (txInfo) {
+          case TX_NAME:
+            cq = TxInfoColumnFamily.TX_NAME_COLUMN;
+            break;
+          case AUTO_CLEAN:
+            cq = TxInfoColumnFamily.AUTO_CLEAN_COLUMN;
+            break;
+          case EXCEPTION:
+            cq = TxInfoColumnFamily.EXCEPTION_COLUMN;
+            break;
+          case RETURN_VALUE:
+            cq = TxInfoColumnFamily.RETURN_VALUE_COLUMN;
+            break;
+          default:
+            throw new IllegalArgumentException("Unexpected TxInfo type " + txInfo);
+        }
+        scanner.fetchColumn(cq.getColumnFamily(), cq.getColumnQualifier());
+
+        return StreamSupport.stream(scanner.spliterator(), false)
+            .map(e -> deserializeTxInfo(txInfo, e.getValue().get())).findFirst().orElse(null);
+      } catch (TableNotFoundException e) {
+        throw new IllegalStateException(tableName + " not found!", e);
+      }
+    }
+
+    @Override
+    public long timeCreated() {
+      verifyReserved(false);
+
+      return scanTx(scanner -> {
+        scanner.setRange(getRow(tid));
+        TxColumnFamily.CREATE_TIME_COLUMN.fetch(scanner);
+        return StreamSupport.stream(scanner.spliterator(), false)
+            .map(e -> Long.parseLong(e.getValue().toString())).findFirst().orElse(0L);
+      });
+    }
+
+    @Override
+    public void push(Repo<T> repo) throws StackOverflowException {
+      verifyReserved(true);
+
+      try {
+
+        // TODO: should we push find top into the mutator?
+        // We also likely need a conditional mutation to make sure we are incrementing the latest
+        Optional<Integer> top = findTop();
+
+        if (top.filter(t -> t >= maxRepos).isPresent()) {
+          throw new StackOverflowException("Repo stack size too large");
+        }
+
+        FateMutator<T> fateMutator = newMutator(tid);
+        fateMutator.putRepo(top.map(t -> t + 1).orElse(1), repo).mutate();
+      } catch (StackOverflowException soe) {
+        throw soe;
+      }
+    }
+
+    @Override
+    public void pop() {
+      verifyReserved(true);
+
+      // TODO: should we push find top into the mutator?
```

Comment:
could remove this TODO

Reply:
I'm going to go ahead and just remove the TODOs and clean things up, I figure we can just create follow on issues for anything we plan to do after such as conditional mutations